embassy-nrf/Cargo.toml

```diff
@@ -47,11 +47,14 @@ embassy = { version = "0.1.0", path = "../embassy" }
 embassy-macros = { version = "0.1.0", path = "../embassy-macros", features = ["nrf"]}
 embassy-hal-common = {version = "0.1.0", path = "../embassy-hal-common" }
 
+embedded-hal-02 = { package = "embedded-hal", version = "0.2.6" }
+embedded-hal-1 = { package = "embedded-hal", version = "1.0.0-alpha.6", git = "https://github.com/embassy-rs/embedded-hal", branch = "async" }
```

Comment:
I find this confusing having to depend on two flavours.

Reply:
You mean 0.2 and 1.0? This is so the HAL can be used with drivers using either 0.2 or 1.0 traits. 

The alternative would be to completely drop 0.2 and tell users to use [embedded-hal-compat](https://github.com/ryankurte/embedded-hal-compat) if they want to use a 0.2 driver. 

I prefer directly supporting both however, because 0.2 is *massively widespread* in the ecosystem, so it'll take years for everything to upgrade to 1.0, so this will be a major usability issue until then.

Reply:
At what point do you think we will let get of 0.2 support? Should we at least mark it as being deprecated?

Reply:
No idea about the timeline, but I guess years.

Not sure if you can mark a trait impl as deprecated, but even if you could it'd be ineffective: the user getting the warning can do nothing about it, it's the driver author that should do the work of updating it.

Reply:
Having gone through the embedded-hal-compat annoyance, I think the approach in this PR is much more helpful to end users 🚀